### ISSUE TITLE
Simplify `_get_field_info` logic

### ISSUE THREAD

**Issue by cphyc** [yt-project/yt#3216]
At the moment the code in `yt/data_objects/static_output.py:_get_field_info` is very circumvoluted, in part due to the support of ambiguous field access.
Once we get rid of this feature™ (yt 4.1.x), we could refactor the function and make it more straightforward to understand.

**Comment by neutrinoceros** [comment]
I agree. In fact I think I already refactored it once for clarity but there's clearly room left for improvement.
The reason I didn't give it a second pass is that I see this as an important part of things that should be reconsidered in the Grand Scheme of a providential redesign of the field system (I think there was a issue where we started a discussion about this with @matthewturk a while back but... I can't find it back :/)
